clusterloader2/pkg/measurement/common/scheduling_throughput.go

```diff
@@ -32,153 +29,86 @@ import (
 	}
 }
 
-func createSchedulingThroughputMeasurement() measurement.Measurement {
-	return &schedulingThroughputMeasurement{}
-}
+type schedulingThroughputGatherer struct{}
 
-type schedulingThroughputMeasurement struct {
-	schedulingThroughputs []float64
-	isRunning             bool
-	stopCh                chan struct{}
-}
+func (a *schedulingThroughputGatherer) Gather(executor QueryExecutor, startTime time.Time, config *measurement.Config) ([]measurement.Summary, error) {
+	throughputSummary, err := a.getThroughputSummary(executor, startTime, config)
 
-// Execute supports two actions:
-// - start - starts the pods scheduling observation.
-//   Pods can be specified by field and/or label selectors.
-//   If namespace is not passed by parameter, all-namespace scope is assumed.
-// - gather - creates summary for observed values.
-func (s *schedulingThroughputMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
-	action, err := util.GetString(config.Params, "action")
 	if err != nil {
 		return nil, err
 	}
-	switch action {
-	case "start":
-		if s.isRunning {
-			klog.V(3).Infof("%s: measurement already running", s)
-			return nil, nil
-		}
-		selector := measurementutil.NewObjectSelector()
-		if err := selector.Parse(config.Params); err != nil {
-			return nil, err
-		}
-		measurmentInterval, err := util.GetDurationOrDefault(config.Params, "measurmentInterval", defaultSchedulingThroughputInterval)
-		if err != nil {
-			return nil, err
-		}
-		s.stopCh = make(chan struct{})
-		return nil, s.start(config.ClusterFramework.GetClientSets().GetClient(), selector, measurmentInterval)
-	case "gather":
-		threshold, err := util.GetFloat64OrDefault(config.Params, "threshold", 0)
-		if err != nil {
-			klog.Warningf("error while getting threshold param: %v", err)
-		}
-		enableViolations, err := util.GetBoolOrDefault(config.Params, "enableViolations", true)
-		if err != nil {
-			klog.Warningf("error while getting enableViolations param: %v", err)
-		}
-		summary, err := s.gather(threshold)
-		if err != nil {
-			if !errors.IsMetricViolationError(err) {
-				klog.Errorf("%s gathering error: %v", config.Identifier, err)
-				return nil, err
-			}
-			if !enableViolations {
-				err = nil
-			}
-		}
-		return summary, err
-	default:
-		return nil, fmt.Errorf("unknown action %v", action)
-	}
-}
-
-// Dispose cleans up after the measurement.
-func (s *schedulingThroughputMeasurement) Dispose() {
-	s.stop()
-}
-
-// String returns a string representation of the measurement.
-func (*schedulingThroughputMeasurement) String() string {
-	return schedulingThroughputMeasurementName
-}
 
-func (s *schedulingThroughputMeasurement) start(clientSet clientset.Interface, selector *measurementutil.ObjectSelector, measurmentInterval time.Duration) error {
-	ps, err := measurementutil.NewPodStore(clientSet, selector)
+	summaryName, err := util.GetStringOrDefault(config.Params, "summaryName", a.String())
```

Comment:
Why can we need a param for this?

Reply:
I vote for removing this and using "schedulingThroughputMeasurementName" as previously.

Reply:
Removed. I was following other prometheus based slos, not sure why they have it.

Reply:
@wojtek-t After discussing with @mborsz I will revert it and add summaryName parameter. It will make it easier to process results in perf-dash.